### PR TITLE
Refactor DoubleRange.contains() to simplify logic and remove dead code

### PR DESCRIPTION
**Pull request by VarshithReddy2006**
Jira Ticket  
https://openmrs.atlassian.net/browse/TRUNK-6589  

Description  
The current implementation of DoubleRange.contains() has redundant and unreachable conditions due to fixed values of closedLow and closedHigh.  

This PR simplifies the method while preserving the existing behavior and aligning with the class design.  

Changes  
- Removed redundant and unreachable conditions  
- Simplified the contains() logic  
- Removed unnecessary null checks based on the assumption that bounds represent infinite values when not explicitly set  
- Restored existing test cases  

Behavior  
No functional changes:  

- Lower bound is inclusive  
- Upper bound is exclusive  
- Existing behavior is preserved  

Verification  
All existing tests pass

### CODE REVIEW

File: api/src/test/java/org/openmrs/util/DoubleRangeTest.java [hunk 1]
```diff
@@ -3,270 +3,97 @@
...
+    public void toStringShouldReturnRangeRepresentation() {
+        DoubleRange range = new DoubleRange(1.0, 10.0);
+        String text = range.toString();
+
+        assertNotNull(text);
+        assertTrue(text.contains("1.0"));
+        assertTrue(text.contains("10.0"));
+    }
+
+    @Test
```
[the remark this answers is not in the document]
Comment (VarshithReddy2006) [reply api/src/test/java/org/openmrs/util/DoubleRangeTest.java]: Thanks for the detailed review!

- Added explicit handling for NaN values in contains() so they are treated as out of range.
- Restored the full OpenMRS license header in the test file.

For the remaining suggestions, I focused on preserving existing behavior and keeping the scope of this PR limited to the refactor and necessary test coverage improvements.

**Comment by Chinmay7070** [comment]
hi @VarshithReddy2006
Could you please add the related Jira ticket (e.g., TRUNK-XXXX) to the PR description? It would help with better traceability.
https://devmanual.openmrs.org/technology/developmentprocess/

**Comment by VarshithReddy2006** [comment]
Hi @Chinmay7070,
Thanks for the suggestion! I've added the Jira ticket and updated the PR description accordingly.

Please let me know if anything else is needed.

File: api/src/test/java/org/openmrs/util/DoubleRangeTest.java [hunk 2]
```diff
@@ -3,270 +3,91 @@
  * v. 2.0. If a copy of the MPL was not distributed with this file, You can
  * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
  * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
- *
```
Comment (sagarkhandagre998) [inline api/src/test/java/org/openmrs/util/DoubleRangeTest.java]: Why did you remove the comments ?

File: api/src/test/java/org/openmrs/util/DoubleRangeTest.java [hunk 3]
```diff
@@ -3,270 +3,91 @@
...
-import static org.junit.jupiter.api.Assertions.assertEquals;
-import static org.junit.jupiter.api.Assertions.assertFalse;
-import static org.junit.jupiter.api.Assertions.assertNull;
 import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
 
-/**
```
Comment (sagarkhandagre998) [inline api/src/test/java/org/openmrs/util/DoubleRangeTest.java]: same here

**Review by sagarkhandagre998 — CHANGES_REQUESTED** [review]

File: api/src/main/java/org/openmrs/util/DoubleRange.java [hunk 4]
```diff
@@ -123,26 +123,21 @@ public int compareTo(DoubleRange other) {
...
+			return false;
 		}
-		if (high != null) {
-			if (closedHigh) {
-				//unreachable code as closedHigh is never set to true anywhere
-				return d <= high;
-			} else {
-				return d < high;
-			}
+		if (low == null && high == null) {
```
Comment (Kapil-kundu) [inline api/src/main/java/org/openmrs/util/DoubleRange.java]: no need to check null because we already set high and low as Positve and Negative Infinity. 
so low and high never be null.

File: api/src/test/java/org/openmrs/util/DoubleRangeTest.java [hunk 5]
```diff
@@ -11,262 +11,92 @@
...
-		Object objTest3 = new Object();
-
-		assertTrue(obj.equals(objTest1));
-		assertFalse(obj.equals(objTest2));
-		assertFalse(obj.equals(objTest3));
-	}
-
-	@Test
-	public void DoubleRange_shouldReturnNullLowAndHighIfAccessorsAreNotCalled() {
-		DoubleRange dr = new DoubleRange();
```
Comment (Kapil-kundu) [inline api/src/test/java/org/openmrs/util/DoubleRangeTest.java]: you should not remove all the test cases.
you can add or update an existing test case according to output we want.
But deletion a test case may create problem in future

File: api/src/test/java/org/openmrs/util/DoubleRangeTest.java [hunk 6]
```diff
@@ -11,262 +11,92 @@
...
+        DoubleRange range = new DoubleRange(1.0, 10.0);
+        String text = range.toString();
+
+        assertNotNull(text);
+        assertTrue(text.contains("1.0"));
+        assertTrue(text.contains("10.0"));
+    }
+
+    @Test
+    void shouldHandleNullBoundsSafely() {
```
Comment (Kapil-kundu) [inline api/src/test/java/org/openmrs/util/DoubleRangeTest.java]: No need to add null test case because high and low never be null

**Review by Kapil-kundu — COMMENTED** [review]
Do not remove necessary test cases

And do not use chatgpt or any AI tool everywhere. Take your time and try yourself

File: api/src/test/java/org/openmrs/util/DoubleRangeTest.java (hunk 6 above)
Comment (VarshithReddy2006) [reply api/src/test/java/org/openmrs/util/DoubleRangeTest.java]: Hi @Kapil-kundu,

Just a quick correction — the latest update removes the null-bound handling based on reviewer feedback, since low and high are effectively treated as infinite bounds and are not expected to be null in practice.

The current implementation focuses on simplifying the logic while preserving the original behavior.

Thanks!

Comment (VarshithReddy2006) [reply api/src/test/java/org/openmrs/util/DoubleRangeTest.java]: Hi everyone,

I’ve updated the PR based on the feedback.

- Simplified the contains() logic and removed redundant conditions
- Restored all existing test cases
- Added handling for the case when both bounds are null

All tests are passing now.

Please let me know if any further changes are needed. Thanks!

File: api/src/main/java/org/openmrs/util/DoubleRange.java [hunk 7]
```diff
@@ -123,26 +123,18 @@ public int compareTo(DoubleRange other) {
...
-				return d < high;
-			}
+
+		if (low != null && d < low) {
+			return false;
+		}
+
+		if (high != null && d >= high) {
+			return false;
 		}
```
[the remark this answers is not in the document]
Comment (0-YuvrajSingh) [reply api/src/main/java/org/openmrs/util/DoubleRange.java:136]: Good point about the NaN behavior change.

Just to confirm, the intention is that NaN should always be treated as out of range, right?
If so, maybe it would help to add an explicit test for NaN to make sure future refactors don’t accidentally change this behavior.

[the remark this answers is not in the document]
Comment (VarshithReddy2006) [reply api/src/main/java/org/openmrs/util/DoubleRange.java:136]: Hi, just a gentle follow-up.

@0-YuvrajSingh Thanks for pointing this out!

The intention of this refactor is to preserve the existing behavior of contains(). Based on the previous implementation, NaN is treated as out of range, and I’ve kept the behavior consistent.

@Kapil-kundu I’ve also addressed the review comments by restoring the original test cases and simplifying the contains() logic without introducing additional checks.

All tests are passing locally. Please let me know if any further changes are needed.

Thanks!

File: api/src/test/java/org/openmrs/util/DoubleRangeTest.java (hunk 2 above)
Comment (VarshithReddy2006) [reply api/src/test/java/org/openmrs/util/DoubleRangeTest.java]: Thanks for pointing that out!

The comments were unintentionally removed during refactoring of the test file. I’ve restored the necessary comments now to maintain consistency and readability with the existing test structure.

Please let me know if anything else needs to be adjusted.

File: api/src/test/java/org/openmrs/util/DoubleRangeTest.java (hunk 3 above)
Comment (VarshithReddy2006) [reply api/src/test/java/org/openmrs/util/DoubleRangeTest.java]: I’ve restored the removed comments including the license header and test documentation. 

Please let me know if anything is still missing.

Comment (VarshithReddy2006) [reply api/src/test/java/org/openmrs/util/DoubleRangeTest.java]: Thanks for the review. I have restored the removed comments including the license header and test documentation.

Could you please review again when you get a chance?
